### PR TITLE
Support for Llama 70-B

### PR DESCRIPTION
**Pull request by AmineDiro**
Hello, 
Solves #402 .

This is a temporary fix for supporting the Llama-2 70B model. I wanted to open a draft PR to get your feedbacks on this implementation for supporting the `n_gqa` params : 
- Added `n_gqa` as a optional param in `ModelParameters`
- Added `LlamaModelVersion` enum akin to other [e_model union in llama.cpp](https://github.com/ggerganov/llama.cpp/blob/8dae7ce68437faf1fa96ec0e7687b8700956ef20/llama.cpp#L76)
- Modified self-attention evaluation to use the `n_head_kv` for `K` and `V` instead of `n_head`

Here is the `llama-2-70B--chat.ggmlv3.q4_0.bin` model loaded on A100 GPU : 
Annotation 2023-08-17 203513

### CODE REVIEW

File: crates/ggml/sys/build.rs [hunk 1]
```diff
@@ -332,6 +332,7 @@ fn enable_cublas(build: &mut cc::Build, out_dir: &Path) {
             .arg("static")
             .arg("--generate-code=arch=compute_52,code=[compute_52,sm_52]")
             .arg("--generate-code=arch=compute_61,code=[compute_61,sm_61]")
+            .arg("--generate-code=arch=compute_75,code=[compute_75,sm_75]")
```
Comment (LLukas22) [inline crates/ggml/sys/build.rs:335]: Is this needed for newer cards?

File: crates/models/llama/src/lib.rs [hunk 2]
```diff
@@ -447,3 +486,13 @@ struct Layer {
...
+
+/// Available Llama models
+enum LlamaModelVersion {
+    Model3b,
+    Model7b,
+    Model13b,
+    Model30b,
+    Model65b,
+    Model70b,
+}
```
Comment (LLukas22) [inline crates/models/llama/src/lib.rs:498]: Version hints at the actual llama model version (1 or 2), maybe rename it into `LlamaModelType`?

**Comment by LLukas22** [comment]
Looks good, some small nitpicks but if the CI passes it should be good to go 👍

File: crates/ggml/sys/build.rs (hunk 1 above)
Comment (AmineDiro) [reply crates/ggml/sys/build.rs:335]: Yes  I had issues with my card RTX2070, I guess we should also support 8.x architectures like A100

File: crates/models/llama/src/lib.rs (hunk 2 above)
Comment (AmineDiro) [reply crates/models/llama/src/lib.rs]: That's true, I changed this in [a16caba] (https://github.com/rustformers/llm/pull/409/commits/a16cabaf5b9b5c1711fe4689e3c00f492524f4e7)

**Comment by AmineDiro** [comment]
@LLukas22  Thanks for the review 👍🏼 !

**Comment by LLukas22** [comment]
Thanks for implementing this :D

**Pull request merged**